src/main/java/com/faforever/client/game/GamesTableController.java

```diff
@@ -87,14 +93,21 @@ public void initializeGameTable(ObservableList<Game> games) {
     passwordProtectionColumn.setCellValueFactory(param -> param.getValue().passwordProtectedProperty());
     passwordProtectionColumn.setCellFactory(param -> passwordIndicatorColumn());
     mapPreviewColumn.setCellFactory(param -> new MapPreviewTableCell(uiService));
-    mapPreviewColumn.setCellValueFactory(param -> new ObjectBinding<Image>() {
+    mapPreviewColumn.setCellValueFactory(param -> new ObjectBinding<Image>() {//TODO: do this async
       {
         bind(param.getValue().mapFolderNameProperty());
       }
 
       @Override
       protected Image computeValue() {
-        return mapService.loadPreview(param.getValue().getMapFolderName(), PreviewSize.SMALL);
+        CompletableFuture<Image> mapPreviewImageFuture = CompletableFuture.supplyAsync(() -> mapService.loadPreview(param.getValue().getMapFolderName(), PreviewSize.SMALL));
+        Image mapPreviewImage = null;
+        try {
+          mapPreviewImage = mapPreviewImageFuture.get();
+        } catch (ExecutionException | InterruptedException e) {
+          logger.error("Could not load map preview", e);
+        }
+        return mapPreviewImage;
```

Comment:
I'm still unhappy with my async image loading solution, it makes working with images pretty difficult. I need a better solution.